src/wordy/ast/BinaryExpressionNode.java

```diff
@@ -59,4 +62,57 @@ public String toString() {
     protected String describeAttributes() {
         return "(operator=" + operator + ')';
     }
+
+    @Override
+    protected double doEvaluate(EvaluationContext context) {
```

Comment:
What I wrote for this method works, but Paul's version is safer. If an invalid operator were to be passed, the method would return 0.0, which could be very bad.